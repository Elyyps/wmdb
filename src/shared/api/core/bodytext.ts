export const generateDummyBodyText = () => `
<div style="padding-bottom: 64px">
<h1 style="margin-top: 64px">I'm the BodyText Module</h1>
<h2>I came from a plein HTML string</h2>
<span style=" backgroundColor: red; min-width: 100%; width: 100%; color: white; text-align: center;">And I support inline style</span> 
</div>
`;
