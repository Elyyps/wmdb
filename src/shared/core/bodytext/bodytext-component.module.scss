.body-text {
  color: $gray;

  h3 {
    margin: 0 0 16px 0;
    + p {
      margin: 0 0 40px 0;
    }
  }

  @media (max-width: $breakpoint-small-max) {
    h3 {
      margin: 0 0 8px 0;
      + p {
        margin: 0 0 32px 0;
      }
    }
  }

  p ~ ul,
  p ~ ol {
    margin-top: -16px;
  }

  a {
    color: rgba(var(--colour-version-website), 1);
    text-decoration: underline !important;
    font-weight: 500;
  }
}

.body-text {
  .body-text-container {
    max-width: $container-max-width;
  }

  .body-text-container-outer {
    max-width: $container-max-width;
    margin: auto;
    display: flex;
  }

  position: relative;
  @media (max-width: $breakpoint-small-max) {
    .uk-grid {
      > div:not(:last-child) {
        margin-bottom: 16px;
      }
    }
  }
  &_foreground {
    position: relative;
  }

  &--left {
    .body-text-container-outer {
      justify-content: flex-start;
    }
  }

  &--center {
    .body-text-container-outer {
      justify-content: center;
    }
  }

  &--right {
    .body-text-container-outer {
      justify-content: flex-end;
    }
  }

  &--large {
    .body-text-container {
      max-width: calc(#{$container-max-width} * 0.8);
    }
  }

  &--medium {
    .body-text-container {
      max-width: calc(#{$container-max-width} * 0.5);
    }
  }

  &--small {
    .body-text-container {
      max-width: calc(#{$container-max-width} * 0.3);
    }
  }
  &--full {
    .body-text-container {
      max-width: $container-max-width;
    }
  }
}
