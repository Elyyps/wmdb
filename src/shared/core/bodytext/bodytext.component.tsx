import * as React from "react";
import styles from "./bodytext-component.module.scss";
import { HtmlRenderComponent } from "../html-render";

export interface IBodyTextComponentProps {
  containerAlignment?: "left" | "center" | "right";
  containerSize?: "large" | "medium" | "small" | "full";
  html: string;
  paddingBottom?: string;
  paddingTop?: string;
  style?: object;
}

export const BodyTextComponent = (props: IBodyTextComponentProps) => {
  const { style, containerAlignment, containerSize, html } = props;

  const alignmentDefault = containerAlignment ? containerAlignment : "left";
  const sizeDefault = containerSize ? containerSize : "full";

  return (
    <div
      style={style}
      className={`uk-container ${styles[`body-text`]} ${styles[`body-text--${alignmentDefault}`]} ${
        styles[`body-text--${sizeDefault}`]
      }`}
    >
      <HtmlRenderComponent html={html} />
    </div>
  );
};
