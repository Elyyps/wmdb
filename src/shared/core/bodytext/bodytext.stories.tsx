import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BodyTextComponent } from "./bodytext.component";

storiesOf("Bodytext", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BodyTextComponent html="<h1>Plein HTML here</h1>" />);
