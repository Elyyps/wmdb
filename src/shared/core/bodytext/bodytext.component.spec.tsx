import React from "react";
import { shallow } from "enzyme";
import { BodyTextComponent } from "./bodytext.component";

describe("[BodyText]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <BodyTextComponent containerAlignment="center" containerSize="full" html="<h1>Plein HTML here</h1>" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
