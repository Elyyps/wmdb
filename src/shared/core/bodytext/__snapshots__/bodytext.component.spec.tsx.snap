// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BodyText] should render component without crashing 1`] = `
<div
  className="uk-container body-text body-text--center body-text--full"
>
  <HtmlRenderComponent
    html=" <h2>
      Hogere kortingen, meer producten en meer uiteenlopende productcategorieÃ«n. Dit gaat er gebeuren met Black Friday.
    </h2>
    <h2>1. Wees er snel bij, winkels beginnen eerder met Black Friday</h2>
    <p>
      Vorig jaar zagen we deze trend al opkomen. Winkels willen altijd graag de eerste zijn met hun deal, daarom
      beginnen een aantal winkels niet op Black Friday, maar vaak een paar dagen eerder. Het is daarom belangrijk dat je
      niet alleen op 23 november 2018 <Link to=\\"#\\">Black Friday Nederland</Link> goed in de gaten houdt, maar ook in de
      week voor de kortingsdag van het jaar de website al iedere dag even checkt. Je weet nooit wanneer de deals online
      komen. Toch zien we wel dat de allerbeste deals pas echt vanaf Black Friday ingaan, maar wees er dan snel bij!
      Want voordat je het weet is jouw favoriete product uitverkocht.
    </p>
    <h2>2. Groots uitpakken met Telecom deals</h2>
    <p>
      Telecom deals zijn niet meer weg te denken van Black Friday. Wij verwachten dat er dit jaar nog meer Telecom deals
      online komen dan vorig jaar. Ben je dus op zoek naar een <Link to=\\"#\\">mobiel</Link>? Wacht dan nog even tot Black
      Friday want we verwachten dat providers als <Link to=\\"#\\">Vodafone</Link>, <Link to=\\"#\\">T-Mobile</Link>,{\\" \\"}
      <Link to=\\"#\\">BEN</Link> en <Link to=\\"#\\">Tele2</Link> weer groots uit gaan pakken met de smartphone deals van 2018.
      De Samsung Galaxy of de Apple Iphone zijn smartphones waar je altijd wel een fikse korting op kunt vinden. Kun je
      niet wachten? Bekijk dan alvast de <Link to=\\"#\\">Samsung Galaxy deals</Link> van dit moment.
    </p>
    <h2>3. 45% meer deelnemende winkels in 2019</h2>
    <ImageComponent src={Oasis} width=\\"279px\\" height=\\"146px\\" className=\\"align-right\\" alt=\\"image description\\" />
    <p>
      Naar verwachting doen er in 2018 bijna de helft meer winkels mee aan <Link to=\\"#\\">Black Friday 2018</Link>.
      Daardoor zal het aantal deals ook bijna met de helft stijgen. Het is dus belangrijk dat je jouw favoriete winkel
      goed in de gaten houdt, want voordat je het weet wordt je ondergesneeuwd door de kortingen en kun je jouw deal
      niet meer vinden.
    </p>
    <h2>4. Meer lokale scherpe aanbiedingen</h2>
    <p>
      Behalve grote namen als <Link to=\\"#\\">adidas</Link>, <Link to=\\"#\\">MediaMarkt</Link>, <Link to=\\"#\\">Ziggo</Link> en{\\" \\"}
      <Link to=\\"#\\">Bol.com</Link> is het ook slim om dit jaar de kleine lokale winkels goed in de gaten te houden. Naar
      verwachting zullen de kleine winkels en boetiekjes in jouw dorp of stad ook meedoen aan Black Friday. Black Friday
      Nederland richt zich echter alleen op de landelijke bekende winkels, dus de aanbiedingen van dat kleine winkeltje
      in jouw dorp zul je zelf in de gaten moeten houden. Dit kan vaak al door de winkel te volgen op Social Media en
      even te vragen of ze dit jaar meedoen aan Black Friday.
    </p>
    <h2>5. Meer producten in reis-, interieur- en fashionbranche</h2>
    <p>
      Dit jaar verwachten wij dat vooral de reisbranche uit gaat pakken. Deze speler was er vorig jaar ook bij met onder
      andere merken als: <Link to=\\"#\\">Ryanair</Link>, Hotels.com, <Link to=\\"#\\">Expedia</Link> en Hilton. De verwachting
      is vooral dat er kortingen komen op vliegtickets en hotels met Black Friday 2018. Houd onze reispagina goed in de
      gaten voor een recent overzicht van alle <Link to=\\"#\\">reisaanbiedingen</Link> met Black Friday.
    </p>
    <p>Een bullet list:</p>
    <ul>
      <li>First</li>
      <li>Second</li>
      <li>Third</li>
      <li>Fourth</li>
    </ul>
    <p>Een numerieke lijst:</p>
    <ol>
      <li>First</li>
      <li>Second</li>
      <li>Third</li>
      <li>Fourth</li>
    </ol>"
  />
</div>
`;
